server/src/internalClusterTest/java/org/elasticsearch/cluster/routing/allocation/decider/DiskThresholdDeciderIT.java

```diff
@@ -151,13 +150,17 @@ public void testHighWatermarkNotExceeded() throws InterruptedException {
         // reduce disk size of node 0 so that no shards fit below the high watermark, forcing all shards onto the other data node
         // (subtract the translog size since the disk threshold decider ignores this and may therefore move the shard back again)
         fileSystemProvider.getTestFileStore(dataNode0Path).setTotalSpace(minShardSize + WATERMARK_BYTES - 1L);
-        refreshDiskUsage();
-        assertThat(getShardRoutings(dataNode0Id), empty());
+        assertBusy(() -> {
+            refreshDiskUsage();
```

Comment:
`refreshDiskUsage` triggers a reroute, shouldn't we have that outside the busy assert and just make sure the the CS application is fully processed by busy asserting on the line below only? That should be all that's needed?

Reply:
I agree, I think you're right.